### PR TITLE
Feature/time comparison

### PR DESCRIPTION
**Pull request by hubert**
- move old Ozone.call into Ozone::Formatter module
- create Ozone::Time class for doing comparisons

### CODE REVIEW

File: lib/ozone.rb [hunk 1]
```diff
@@ -3,17 +3,49 @@
...
+    def call(time:, offset:, observes_dst: true, format: '%Y-%m-%d %H:%M')
       timezone = ActiveSupport::TimeZone[offset/60]
-      2.hours.since(time.in_time_zone(timezone)).strftime(format)
-    else
-      time_with_zone.strftime(format)
+      time_with_zone = time.in_time_zone(timezone)
+      if time_with_zone.dst? && !observes_dst
+        offset -= 120
+        timezone = ActiveSupport::TimeZone[offset/60]
+        2.hours.since(time.in_time_zone(timezone)).strftime(format)
```
Comment (l8nite) [inline lib/ozone.rb]: It'd be good to have a comment here explaining the "magic"

File: lib/ozone.rb [hunk 2]
```diff
@@ -3,17 +3,49 @@
...
+      else
+        time_with_zone.strftime(format)
+      end
+    end
+  end
+
+  class Time
```
Comment (l8nite) [inline lib/ozone.rb]: one class/module per file

**Comment by l8nite** [comment]
:diamond_shape_with_a_dot_inside:

File: lib/ozone.rb (hunk 1 above)
Comment (hubert) [inline lib/ozone.rb]: this implies that i can explain the magic ;-)

File: lib/ozone.rb (hunk 2 above)
Comment (hubert) [inline lib/ozone.rb]: i agree with this generally, but in this situation i feel it clearer in one single file given the interconnectedness and number of classes/modules.

if you or others feel strongly, i'll relent.

File: README.md [hunk 3]
```diff
@@ -23,16 +26,54 @@ Or install it yourself as:
 
 ## Usage
```
Comment (ak47) [inline README.md]: time_with_dst isn't defined

File: README.md [hunk 4]
```diff
@@ -23,16 +26,54 @@ Or install it yourself as:
...
+> ozone_time.before?(1.second.until(dst_time))
+=> false
+```
+
+However, when observes_dst is false:
 
 ```ruby
-> Ozone.call(time: t, offset: -480, observes_dst: true)
+ozone_time = Time.new(
+  time: time_with_dst,
```
Comment (ak47) [inline README.md]: time_with_dst isn't defined

File: spec/ozone_spec.rb [hunk 5]
```diff
@@ -1,52 +1,151 @@
...
+            time: time_without_dst,
+            offset: -480,
+            observes_dst: false,
+          )
+          expect(adjusted_time).to eq("2014-03-09 01:59")
+        end
+      end
+
+      context 'time is during daylight savings' do
+        it 'converts a utc time to time with time zone and makes no dst adustment' do
```
Comment (ak47) [inline spec/ozone_spec.rb]: doesn't this make a dot adjustment?

Comment (hubert) [inline spec/ozone_spec.rb]: no adjustment based on the fact that it is during DST and we're observing daylight savings, so we leave the time the same.

File: README.md (hunk 4 above)
Comment (hubert) [inline README.md]: you're like a human interpreter :dromedary_camel:

**Pull request merged**